crates/bevy_ecs/src/lib.rs

```diff
@@ -637,8 +637,22 @@ mod tests {
     #[test]
     fn table_add_remove_many() {
         let mut world = World::default();
-        let mut entities = Vec::with_capacity(10_000);
-        for _ in 0..1000 {
+        #[cfg(miri)]
+        let mut entities = Vec::with_capacity(10);
+        #[cfg(not(miri))]
+        let mut entities = Vec::with_capacity(1_000);
+
+        let to;
+        #[cfg(miri)]
+        {
+            to = 10;
+        }
+        #[cfg(not(miri))]
+        {
+            to = 1000;
+        }
```

Comment:
```suggestion
      #[cfg(miri)]
      let (mut entities, to) = {
          let to = 10;
          (Vec::with_capacity(to), to)
      };
      #[cfg(not(miri))]
      let (mut entities, to) = {
          let to = 10_000;
          (Vec::with_capacity(to), to)
      };
```

alternatively, would using [`Vec::capacity()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.capacity) instead of `to` in the for loop below work?